.buildkite/pipeline.yml

```diff
@@ -9,22 +9,33 @@ env:
 
 steps:
   - label: ":face_with_peeking_eye: Lint"
-    command: "make lint"
+    command: ".buildkite/run_linter.sh"
+    env:
+      PYTHON_VERSION: "3.10" # We don't really care about linter on multiple versions for now
     timeout_in_minutes: 5
 
-  - label: ":pytest: Test"
-    command: "make test"
+  - label: ":pytest: Test with Python {{ matrix }}"
+    command: ".buildkite/run_tests.sh"
+    env:
+      PYTHON_VERSION: "{{ matrix }}"
     timeout_in_minutes: 5
+    matrix:
+      - "3.10"
+      - "3.11"
 
   - label: ":shipit: Smoke test"
-    command: "make ftest NAME=dir DATA_SIZE=small"
+    command: ".buildkite/run_functional_test.sh"
+    env:
+      PYTHON_VERSION: "3.10" # TODO: can actually do a matrix too
```

Comment:
```suggestion
      PYTHON_VERSION: "3.11" # TODO: can actually do a matrix too
```

Reply:
Though, I might be interested in doing the full matrix for the smoketest, since they'll run in parallel, right?